presto/dataset.py

```diff
@@ -61,21 +65,50 @@ def __len__(self):
         return self.df.shape[0]
 
     @staticmethod
-    def target_crop(row_d: Dict) -> int:
-        # by default, we predict crop vs non crop
-        return int(row_d["LANDCOVER_LABEL"] == 11)
+    def target_crop(
```

Comment:
nit: this probably makes sense to move into the `WorldCerealLabelledDataset` since its now tightly coupled to attributes in the labelled dataset

Reply:
not really sure how to do that correctly... this method is used in `row_to_arrays`, which is defined in `WorldCerealBase`, and then inherited by all the child classes...
so moving the `target_crop` away would require to rework this `row_to_arrays` too, right?

Reply:
I guess indeed it makes sense to remove `target_function` from `row_to_arrays` in order for the latter to focus only on the input data for presto. The return value of that method than no longer includes the target. In subclasses that need a target (maybe even only `WorldCerealLabelledDataset`) you'd call `row_to_arrays` to collect the inputs and then use `target_function` to assemble the target. The added value here is also that in the case we don't have a ground truth for a row, we don't have to put in hacks to make it work.

Reply:
The labelled dataset `__getitem__` function goes from:

```python
    def __getitem__(self, idx):
        # Get the sample
        df_index = self.indices[idx]
        row = self.df.iloc[df_index, :]
        eo, mask_per_token, latlon, month, valid_month, target = self.row_to_arrays(
            row, self.task_type, self.croptype_list, self.model_mode
        )
        mask_per_variable = np.repeat(mask_per_token, BAND_EXPANSION, axis=1)
        return (
            self.normalize_and_mask(eo),
            target,
            np.ones(self.NUM_TIMESTEPS) * (DynamicWorld2020_2021.class_amount),
            latlon,
            month,
            valid_month,
            mask_per_variable,
```

Now, rows_to_arrays no longer returns a target, so __getitem__ neds to get it:

```python
    def __getitem__(self, idx):
        # Get the sample
        df_index = self.indices[idx]
        row = self.df.iloc[df_index, :]
        eo, mask_per_token, latlon, month, valid_month = self.row_to_arrays(
            row, self.task_type, self.croptype_list, self.model_mode
        )
        target = self.target_function(...)  # new
        mask_per_variable = np.repeat(mask_per_token, BAND_EXPANSION, axis=1)
        return (
            self.normalize_and_mask(eo),
            target,
            np.ones(self.NUM_TIMESTEPS) * (DynamicWorld2020_2021.class_amount),
            latlon,
            month,
            valid_month,
            mask_per_variable,
```